types/FluentJSONSchema.d.ts

```diff
@@ -198,63 +195,75 @@ type MixedSchema7<T> = T extends [
       InferSchemaMap[Y extends TYPE ? Y : never] &
       InferSchemaMap[Z extends TYPE ? Z : never] &
       InferSchemaMap[A extends TYPE ? A : never]
-  : never
+  : never;
 
-export type MixedSchema<T> =
+type MixedSchema<T> =
   | MixedSchema1<T>
   | MixedSchema2<T>
   | MixedSchema3<T>
   | MixedSchema4<T>
   | MixedSchema5<T>
   | MixedSchema6<T>
-  | MixedSchema7<T>
+  | MixedSchema7<T>;
 
 interface SchemaOptions {
-  schema: object
-  generateIds: boolean
+  schema: object;
+  generateIds: boolean;
 }
 
 interface PatternPropertiesOptions {
-  [key: string]: JSONSchema
+  [key: string]: JSONSchema;
 }
 
 interface DependenciesOptions {
-  [key: string]: JSONSchema[]
+  [key: string]: JSONSchema[];
 }
 
-type Key<T> = keyof T | (string & {})
+type Key<T> = keyof T | (string & {});
 
-type DependentSchemaOptions<T extends Partial<Record<string, JSONSchema>>> = Partial<Record<keyof T, JSONSchema>>
+type DependentSchemaOptions<T extends Partial<Record<string, JSONSchema>>> =
+  Partial<Record<keyof T, JSONSchema>>;
 
-type DependentRequiredOptions<T extends Partial<Record<string, string[]>>> = Partial<Record<keyof T, string[]>>
-
-export function withOptions<T>(options: SchemaOptions): T
+type DependentRequiredOptions<T extends Partial<Record<string, string[]>>> =
+  Partial<Record<keyof T, string[]>>;
 
 type ObjectPlaceholder = Record<string | number | symbol, any>;
 
-export interface S extends BaseSchema<S> {
-  string: () => StringSchema
-  number: () => NumberSchema
-  integer: () => IntegerSchema
-  boolean: () => BooleanSchema
-  array: () => ArraySchema
-  object: <T extends ObjectPlaceholder = ObjectPlaceholder>() => ObjectSchema<T>
-  null: () => NullSchema
-  mixed: <
-    T extends
-      | [TYPE]
-      | [TYPE, TYPE]
-      | [TYPE, TYPE, TYPE]
-      | [TYPE, TYPE, TYPE, TYPE]
-      | [TYPE, TYPE, TYPE, TYPE, TYPE]
-      | [TYPE, TYPE, TYPE, TYPE, TYPE, TYPE]
-      | [TYPE, TYPE, TYPE, TYPE, TYPE, TYPE, TYPE]
-  >(
-    types: T
-  ) => MixedSchema<T>
-  raw: (fragment: any) => S
-  FORMATS: FORMATS
+declare namespace s {
+  export class FluentSchemaError extends Error {
+    name: string;
+  }
+
+  interface S extends BaseSchema<S> {
+    string: () => StringSchema;
+    number: () => NumberSchema;
+    integer: () => IntegerSchema;
+    boolean: () => BooleanSchema;
+    array: () => ArraySchema;
+    object: <
+      T extends ObjectPlaceholder = ObjectPlaceholder
+    >() => ObjectSchema<T>;
+    null: () => NullSchema;
+    mixed: <
+      T extends
+        | [TYPE]
+        | [TYPE, TYPE]
+        | [TYPE, TYPE, TYPE]
+        | [TYPE, TYPE, TYPE, TYPE]
+        | [TYPE, TYPE, TYPE, TYPE, TYPE]
+        | [TYPE, TYPE, TYPE, TYPE, TYPE, TYPE]
+        | [TYPE, TYPE, TYPE, TYPE, TYPE, TYPE, TYPE]
+    >(
+      types: T
+    ) => MixedSchema<T>;
+    raw: (fragment: any) => S;
+    FORMATS: FORMATS;
+  }
+
+  export function withOptions<T>(options: SchemaOptions): T;
+
+  export const FluentSchema: S;
```

Comment:
No named export

Reply:
I cannot understand why not, I've seem similar approach in other places. Can you please clarify?

Reply:
But removing this leads to broken import:
 
```typescript
import S, { FluentSchemaError } from '..'
```

Reply:
It should be importing the line below, not named one.

Reply:
```suggestion
  const FluentSchema: S;
```